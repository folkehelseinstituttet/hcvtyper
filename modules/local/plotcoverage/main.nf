process PLOTCOVERAGE {
    tag "$meta.id"
    label 'process_single'

    // Environment with R tidyverse and seqinr packages from the conda-forge channel. Created using seqera containers.
    // URLs:
    // Docker image: https://wave.seqera.io/view/builds/bd-5358395134867368_1?_gl=1*1vaclhd*_gcl_au*MTkxMjgxNTMwMi4xNzUzNzczOTQz
    // Singularity image: https://wave.seqera.io/view/builds/bd-0225dab2b8112adf_1?_gl=1*111m8r6*_gcl_au*MTkxMjgxNTMwMi4xNzUzNzczOTQz
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/b5/b5d7b977f4b94903794f65bbd852248eef87ce608c5eef34d605afabf514f397/data':
        'community.wave.seqera.io/library/r-seqinr_r-tidyverse:5358395134867368' }"

    input:
    tuple val(meta), path(depth)

    output:
    path("*.png")      , emit: png, optional: true
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    bam_coverage.R "${depth}" "${prefix}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        tidyverse: \$(Rscript -e "library(tidyverse); cat(as.character(packageVersion('tidyverse')))")
        seqinr: \$(Rscript -e "library(seqinr); cat(as.character(packageVersion('seqinr')))")
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    # Create a dummy PNG file to mimic the output
    touch ${prefix}.dummy.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        tidyverse: \$(Rscript -e "library(tidyverse); cat(as.character(packageVersion('tidyverse')))")
        seqinr: \$(Rscript -e "library(seqinr); cat(as.character(packageVersion('seqinr')))")
    END_VERSIONS
    """
}
