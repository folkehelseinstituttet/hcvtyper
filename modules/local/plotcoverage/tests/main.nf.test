nextflow_process {

    name "Test Process PLOTCOVERAGE"
    script "../main.nf"
    process "PLOTCOVERAGE"

    tag "modules"
    tag "plotcoverage"

    test("hcv - depth tsv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'toy' ], // meta map
                    file("${projectDir}/modules/local/plotcoverage/tests/toy.3a_D17763_major.major.nodup.tsv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.png[0]).name
                ).match() }
            )
        }

    }

    test("hcv - depth tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'toy' ], // meta map
                    file("${projectDir}/modules/local/plotcoverage/tests/toy.3a_D17763_major.major.nodup.tsv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.png[0]).name
                ).match() }
            )
        }

    }

}
