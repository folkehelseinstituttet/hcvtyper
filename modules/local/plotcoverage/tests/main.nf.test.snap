{
    "hcv - depth tsv - stub": {
        "content": [
            [
                "versions.yml:md5,8f76ebb71c1e290c493f125e17b9d739"
            ],
            "toy.dummy.png"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-14T19:09:37.78441706"
    },
    "hcv - depth tsv": {
        "content": [
            [
                "versions.yml:md5,8f76ebb71c1e290c493f125e17b9d739"
            ],
            "toy.major.3a_D17763.png"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-14T19:09:27.37461944"
    }
}